src/main/java/com/squareup/catalog/demo/util/CatalogObjects.java

```diff
@@ -34,70 +34,59 @@
-   * @param cents the fixed price in cents
+   * @param cents      the fixed price in cents
    * @return the new {@link CatalogObject} that wraps the item variation
    */
   public static CatalogObject item(String clientId, String name, String categoryId, long cents) {
-    return item(clientId, name, categoryId,
-        itemVariation("#" + UUID.randomUUID().toString(), "Regular", cents));
+    return item(clientId, name, categoryId, itemVariation("#" + UUID.randomUUID().toString(), "Regular", cents));
   }
 
   /**
    * Builds an item with the specified variations.
    *
-   * @param clientId the temporary client-generated id
-   * @param name the name of the item variation
-   * @param categoryId the ID of the category that this item belongs to
+   * @param clientId       the temporary client-generated id
+   * @param name           the name of the item variation
+   * @param categoryId     the ID of the category that this item belongs to
    * @param itemVariations the item variations in the item
    * @return the new {@link CatalogObject} that wraps the item variation
    */
-  public static CatalogObject item(String clientId, String name, String categoryId,
-      CatalogObject... itemVariations) {
-    return new CatalogObject()
-        .type(TypeEnum.ITEM)
-        .id(clientId)
-        .presentAtAllLocations(true)
-        .itemData(new CatalogItem()
-            .name(name)
-            .categoryId(categoryId)
-            .variations(asList(itemVariations)));
+  public static CatalogObject item(String clientId, String name, String categoryId, CatalogObject... itemVariations) {
+    return new CatalogObject.Builder(CatalogObjectTypes.ITEM.toString(), clientId).presentAtAllLocations(true)
+        .itemData(
+            new CatalogItem.Builder().name(name).categoryId(categoryId).variations(asList(itemVariations)).build())
+        .build();
   }
 
   /**
    * Builds a fixed price item variation that is available at all locations.
    *
    * @param clientId the temporary client-generated id
-   * @param name the name of the item variation
-   * @param cents the fixed price in cents
+   * @param name     the name of the item variation
+   * @param cents    the fixed price in cents
    * @return the new {@link CatalogObject} that wraps the item variation
    */
   public static CatalogObject itemVariation(String clientId, String name, long cents) {
-    return new CatalogObject()
-        .type(TypeEnum.ITEM_VARIATION)
-        .id(clientId)
-        .presentAtAllLocations(true)
-        .itemVariationData(new CatalogItemVariation()
-            .name(name)
-            .pricingType(PricingTypeEnum.FIXED_PRICING)
-            .priceMoney(Moneys.usd(cents)));
+    return new CatalogObject.Builder(CatalogObjectTypes.ITEM_VARIATION.toString(), clientId).presentAtAllLocations(true)
+        .itemVariationData(new CatalogItemVariation.Builder().name(name).pricingType("FIXED_PRICING")
```

Comment:
I'm surprised we don't have Enums for things like pricingTypes and discountTypes anymore :-/

Reply:
Agreed, feels so messy. If the user wants to write clean code, he/she probably have to create their own enums, which is annoying